### PR TITLE
Rename schemas, change meaning of BenchmarkResultCreate

### PR DESCRIPTION
**Pull request by jgehrcke**
For #961.

### CODE REVIEW

File: conbench/api/benchmarks.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 from ..api._endpoint import ApiEndpoint, maybe_login_required
 from ..entities._entity import NotFound
 from ..entities.benchmark_result import (
-    BenchmarkFacadeSchema,
+    BenchmarkResultFacadeSchema,
```
Comment (jgehrcke) [inline conbench/api/benchmarks.py]: Note that personally I am not sure of the information content in `Facade`. I kept it here for now, although maybe it is not useful in the future. Does this try to indicate the JSON surface?

File: conbench/entities/benchmark_result.py [hunk 2]
```diff
@@ -300,7 +300,7 @@ def update(self, data):
 s.Index("benchmark_result_timestamp_index", BenchmarkResult.timestamp)
 
 
-class BenchmarkResultCreate(marshmallow.Schema):
+class BenchmarkResultStatsSchema(marshmallow.Schema):
```
Comment (jgehrcke) [inline conbench/entities/benchmark_result.py:303]: This diff line is the actual and rather significant motivator for this patch. The current state is a little too mind-boggling; and this here also is a necessary change towards implementing a clear distinction between "benchmark" and "benchmark result".

File: conbench/entities/benchmark_result.py [hunk 3]
```diff
@@ -358,57 +358,53 @@ class BenchmarkResultCreate(marshmallow.Schema):
...
         required=False,
         metadata={
-            "description": "The inter-quartile range from `data`, will be calculdated on the server if not present (the preferred method), but can be overridden if sent. Will be marked `null` if any iterations are missing."
+            "description": "The inter-quartile range from `data`, will be calculated on the server if not present (the preferred method), but can be overridden if sent. Will be marked `null` if any iterations are missing."
         },
     )
 
 
-class BenchmarkResultSchema:
-    create = BenchmarkResultCreate()
```
Comment (jgehrcke) [inline conbench/entities/benchmark_result.py:409]: What is/was the value of this wrapping technique?

File: conbench/entities/benchmark_result.py [hunk 4]
```diff
@@ -573,7 +569,7 @@ class _BenchmarkFacadeSchemaCreate(marshmallow.Schema):
             )
         },
     )
-    stats = marshmallow.fields.Nested(BenchmarkResultSchema().create, required=False)
+    stats = marshmallow.fields.Nested(BenchmarkResultStatsSchema(), required=False)
```
Comment (jgehrcke) [inline conbench/entities/benchmark_result.py:572]: Do we lose anything by taking the indirection out?

File: conbench/entities/benchmark_result.py [hunk 5]
```diff
@@ -686,6 +682,6 @@ class _BenchmarkFacadeSchemaUpdate(marshmallow.Schema):
     )
 
 
-class BenchmarkFacadeSchema:
-    create = _BenchmarkFacadeSchemaCreate()
-    update = _BenchmarkFacadeSchemaUpdate()
+class BenchmarkResultFacadeSchema:
+    create = _BenchmarkResultCreateSchema()
+    update = _BenchmarkResultUpdateSchema()
```
Comment (jgehrcke) [inline conbench/entities/benchmark_result.py:687]: Here I again tried to stay as consistent as possible with previous naming, but as I said above, the "facade" term is not one that I am too convinced about yet.

File: conbench/tests/api/_expected_docs.py [hunk 6]
```diff
@@ -978,7 +978,7 @@
             },
         },
         "schemas": {
-            "BenchmarkCreate": {
+            "BenchmarkResultCreate": {
```
Comment (jgehrcke) [inline conbench/tests/api/_expected_docs.py:981]: This is important outfall.

File: conbench/tests/api/_expected_docs.py [hunk 7]
```diff
@@ -1022,7 +1022,7 @@
                         "description": "Reason for the Run (optional, does not need to be unique). Ignored when Run was previously created.",
                         "type": "string",
                     },
-                    "stats": {"$ref": "#/components/schemas/BenchmarkResultCreate"},
+                    "stats": {"$ref": "#/components/schemas/BenchmarkResultStats"},
```
Comment (jgehrcke) [inline conbench/tests/api/_expected_docs.py:1025]: This is a user-facing (well, external dev-facing) clarification and un-confusion.

**Review by jonkeane — APPROVED** [review]
I'm fine with all of this — I don't think anyone is relying on these names specifically (and correct me if I'm wrong, but doesn't actually change any keys or names in the API itself — just what we refer to them here that then gets bubbled up into the docs, yeah?)

😅 that swagger not letting one deep-link to these also means there are no broken URLs after this (not that that would be a huge deal in this case, but if it doesn't work, it can't break :P)

**Comment by jgehrcke** [comment]
> just what we refer to them here that then gets bubbled up into the docs, yeah?

Yes! Thanks for the quick review and opinion.

File: conbench/entities/benchmark_result.py (hunk 3 above)
Comment (austin3dickey) [reply conbench/entities/benchmark_result.py:409]: I suspect it was reserving a spot for a future `update` schema. Not very important

**Pull request merged**